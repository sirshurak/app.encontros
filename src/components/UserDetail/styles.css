#user-detail .img-box figure {
    margin: 0;
    margin-bottom: 5px;
}

#user-detail .img-box figure img {
    width: 120px;
    height: 120px;
    border-radius: 50%;
    margin: 10px 0;
    border-color: white;
    border-style: solid;
    background-color: #d1d1d1;
}

#user-detail .img-box figure .bg {
    position: absolute;
    top: 0px;
    left: 0px;
    width: 100%;
    height: 145px;
    opacity: 0.25;
    background-size: 100% 145px;
    background-color: #d1d1d1;
}

#user-detail .more-info {
    display: grid;
    grid-template-columns: 1fr 1fr;
}

#user-detail .more-info div {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
}

#user-detail .more-info div p {
    margin: 0;
}