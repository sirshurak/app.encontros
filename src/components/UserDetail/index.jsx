import React from "react";

import "./styles.css";

function UserDetail({ user }) {
  return (
    <section id="user-detail">
      <div className="img-box">
        <figure>
          <img src={user?.photo} alt="" />
          <div
            className="bg"
            style={{ backgroundImage: `url("${user?.photo}")` }}
          />
        </figure>
      </div>
      <h2>{user?.username}</h2>
      <p>{user?.city}</p>
      <div className="more-info">
        <div>
          <p>{user?.gender}</p>
          <p>{user?.sexualOrientation}</p>
        </div>
        <div>
          <p>{user?.age}</p>
        </div>
      </div>
    </section>
  );
}

export default UserDetail;
