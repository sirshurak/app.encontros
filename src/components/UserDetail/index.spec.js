import { shallow, mount } from 'enzyme';
import UserDetail from './index';

const user = {
    photo: 'https://picsum.photos/200/?image=26',
    username: 'username',
    city: 'Test City',
    gender: 'Man',
    sexualOrientation: 'Heterosexual',
    age: 27
}

describe('UserDetail component', () => {
    it('should match shallow snapshop', () => {
        const component = shallow(<UserDetail user={user}/>)
        expect(component).toMatchSnapshot();
    });
});