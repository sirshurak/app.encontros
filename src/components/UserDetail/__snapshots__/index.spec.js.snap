// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDetail component should match shallow snapshop 1`] = `
<section
  id="user-detail"
>
  <div
    className="img-box"
  >
    <figure>
      <img
        alt=""
        src="https://picsum.photos/200/?image=26"
      />
      <div
        className="bg"
        style={
          Object {
            "backgroundImage": "url(\\"https://picsum.photos/200/?image=26\\")",
          }
        }
      />
    </figure>
  </div>
  <h2>
    username
  </h2>
  <p>
    Test City
  </p>
  <div
    className="more-info"
  >
    <div>
      <p>
        Man
      </p>
      <p>
        Heterosexual
      </p>
    </div>
    <div>
      <p>
        27
      </p>
    </div>
  </div>
</section>
`;
